documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/TrashStorageImpl.java

```diff
@@ -328,11 +332,58 @@ public void removeDeadSymlinks(Node node) throws RepositoryException {
     removeDeadSymlinks(node, true);
   }
 
+  @Override
+  public List<Node> getTrashElements(TrashElementNodeFilter trashElementNodeFilter) throws RepositoryException {
+    String mappedSortField = switch (trashElementNodeFilter.getSortField()) {
+      case NAME -> EXO_TITLE;
+      default -> EXO_DATE_MODIFIED;
+    };
+    String sortDirection = trashElementNodeFilter.isAscending() ? "ASC" : "DESC";
+    // all trash elements has the property exo:restorePath, it is added during the move te trash
+    StringBuilder sb = new StringBuilder(SELECT_FROM_TRASH_QUERY);
+    sb.append(" ORDER BY ").append(mappedSortField).append(" ").append(sortDirection);
+
+    List<Node> result = new ArrayList<>();
+
+    ManageableRepository manageableRepository = repositoryService.getCurrentRepository();
+    SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+    Session session = sessionProvider.getSession(this.trashWorkspace, manageableRepository);
+    QueryManager queryManager = session.getWorkspace().getQueryManager();
+
+    QueryImpl query = (QueryImpl) queryManager.createQuery(sb.toString(), Query.SQL);
+    query.setLimit(trashElementNodeFilter.getLimit());
+    query.setOffset(trashElementNodeFilter.getOffset());
+    QueryResult queryResult = query.execute();
+
+    NodeIterator iter = queryResult.getNodes();
+    while (iter.hasNext()) {
+      result.add(iter.nextNode());
+    }
+    return result;
+  }
+
   @Override
   public List<Node> getAllLinks(Node targetNode, String linkType) {
     return getAllLinks(targetNode, linkType, sessionProviderService.getSessionProvider(null));
   }
 
+  @Override
+  public int countDeletedDocuments() {
+    try {
+      StringBuilder sb = new StringBuilder(SELECT_FROM_TRASH_QUERY);
+      ManageableRepository manageableRepository = repositoryService.getCurrentRepository();
+      SessionProvider sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+      Session session = sessionProvider.getSession(this.trashWorkspace, manageableRepository);
+      QueryManager queryManager = session.getWorkspace().getQueryManager();
+      Query query = queryManager.createQuery(sb.toString(), Query.SQL);
+      QueryResult queryResult = query.execute();
+      return (int) queryResult.getRows().getSize();

```

Comment:
I'm afraid of performances of this query.
We know that, when you have millions of children for a node, the JCR have very poor performances.

I would prefer something like this : 
```suggestion
      NodeIterator iter = queryResult.getNodes();
      return iter.getSize();
```

Reply:
done!